packages/mui-utils/src/useDeferredValueCompat/useDeferredValueCompat.ts

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import {
+  nonBlockingBatchCallback,
+  enqueueStateUpdate,
+  runWithBatch,
+} from '../CompatTransitionManager';
+
+function useDeferredValueCompatImpl18<T>(value: T): T {
+  // TODO: On React 18 and higher, using the compat hook is unnecessary,
+  // so we should display a warning.
+  return React.useDeferredValue(value);
+}
```

Comment:
Would a warning really be appropriate? IIUC, the goal of this shim would be to allow material-ui to work with both 17 and 18, which is under the user's control.

Reply:
Yeah, I comment that we can follow what we do in the `useId` hook from @mui/utils.